extensions/sql-migration/src/api/utils.ts

```diff
@@ -261,3 +256,545 @@ export function clearDialogMessage(dialog: window.Dialog): void {
+}
+
+export async function getAzureResourceGroupsDropdownValues(resourceGroups: azureResource.AzureResourceResourceGroup[]): Promise<azdata.CategoryValue[]> {
+	let resourceGroupValues: azdata.CategoryValue[] = [];
+	resourceGroups.forEach((rg) => {
+		resourceGroupValues.push({
+			name: rg.id,
+			displayName: rg.name
+		});
+	});
+	if (resourceGroupValues.length === 0) {
+		resourceGroupValues = [
+			{
+				displayName: constants.RESOURCE_GROUP_NOT_FOUND,
+				name: ''
+			}
+		];
+	}
+	return resourceGroupValues;
+}
+
+export async function getManagedInstances(account?: azdata.Account, subscription?: azureResource.AzureResourceSubscription): Promise<azureResource.AzureSqlManagedInstance[]> {
+	let managedInstances: azureResource.AzureSqlManagedInstance[] = [];
+	try {
+		if (account && subscription) {
+			managedInstances = await getAvailableManagedInstanceProducts(account, subscription);
+		}
+	} catch (e) {
+		console.log(e);
+		managedInstances = [];
+	}
+	managedInstances.sort((a, b) => a.name.localeCompare(b.name));
+	return managedInstances;
+}
+
+export async function getManagedInstancesDropdownValues(managedInstances: azureResource.AzureSqlManagedInstance[], location: azureResource.AzureLocation, resourceGroup: azureResource.AzureResourceResourceGroup): Promise<azdata.CategoryValue[]> {
+	let managedInstancesValues: azdata.CategoryValue[] = [];
+	if (location && resourceGroup) {
+		managedInstances.forEach((managedInstance) => {
+			if (managedInstance.location.toLowerCase() === location.name.toLowerCase() && managedInstance.resourceGroup?.toLowerCase() === resourceGroup.name.toLowerCase()) {
+				let managedInstanceValue: azdata.CategoryValue;
+				if (managedInstance.properties.state === 'Ready') {
+					managedInstanceValue = {
+						name: managedInstance.id,
+						displayName: managedInstance.name
+					};
+				} else {
+					managedInstanceValue = {
+						name: managedInstance.id,
+						displayName: constants.UNAVAILABLE_TARGET_PREFIX(managedInstance.name)
+					};
+				}
+				managedInstancesValues.push(managedInstanceValue);
+			}
+		});
+	}
+
+	if (managedInstancesValues.length === 0) {
+		managedInstancesValues = [
+			{
+				displayName: constants.NO_MANAGED_INSTANCE_FOUND,
+				name: ''
+			}
+		];
+	}
+	return managedInstancesValues;
+}
+
+export async function getVirtualMachines(account?: azdata.Account, subscription?: azureResource.AzureResourceSubscription): Promise<SqlVMServer[]> {
+	let virtualMachines: SqlVMServer[] = [];
+	try {
+		if (account && subscription) {
+			virtualMachines = (await getAvailableSqlVMs(account, subscription)).filter((virtualMachine) => {
+				if (virtualMachine.properties.sqlImageOffer) {
+					return virtualMachine.properties.sqlImageOffer.toLowerCase().includes('-ws'); //filtering out all non windows sql vms.
+				}
+				return true; // Returning all VMs that don't have this property as we don't want to accidentally skip valid vms.
+			});
+		}
+	} catch (e) {
+		console.log(e);
```

Comment:
can you use the telemetry logError functionality instead of using console.log(e)